content/tutorials/custom-zk-chain/20.customizing-your-chain.md

```diff
@@ -0,0 +1,374 @@
+  const tokenSymbol = m.getParameter("symbol", "ZKBT");
+
+  const baseToken = m.contract("CustomBaseToken", [tokenName, tokenSymbol]);
+
+  return { baseToken };
+});
+
+export default CustomBaseTokenModule;
+```
+
+To deploy the token contract, run:
+
+```bash
+npx hardhat ignition deploy ./ignition/modules/CustomBaseToken.ts --network localRethNode
+```
+
+Select `y` to confirm to deploy the contract to the `localRethNode`.
+After deploying, the token contract address should be logged in your console.
+
+Let's verify that the tokens were minted to the deployer address using the Foundry `cast` CLI:
+
+```bash
+cast balance --erc20 <0xYOUR_TOKEN_ADDRESS> 0x36615Cf349d7F6344891B1e7CA7C72883F5dc049 \
+--rpc-url http://localhost:8545
+```
+
+Now that you have some tokens, the next step is to send some to the governor's address on the L1 so that you can register and deploy your chain.
+You can find the address for the governor in `my_elastic_chain/configs/wallets.yaml` under `governor`.
+
+Run the command below to use `cast` to transfer some of your ERC20 tokens to the governor's address on the L1.
+
+```bash
+cast send 0xYOUR_TOKEN_ADDRESS "transfer(address,uint256)" \
+<0x_YOUR_GOVERNOR_ADDRESS> 5000000000000000000 \
+--private-key 0x7726827caac94a7f9e1b160f7ea819f172f7b6f9d2a97f992c38edeab82d4110 \
+--rpc-url http://localhost:8545 \
+--gas-price 30000000000
+```
+
+To verify that this worked, check the balance of the governor's address:
+
+```bash
+cast balance --erc20 <0xYOUR_TOKEN_ADDRESS> \
+<0x_YOUR_GOVERNOR_ADDRESS> \ 
+--rpc-url http://localhost:8545
+```
+
+## Creating a new chain
+
+Now that your ERC20 token is deployed and the governor's address is funded with that token, you can create a new chain.
+
+First, shut down the node server running for `zk_chain_1`.
+
+Move back into the `my_elastic_chain` ecosystem folder and run the `chain create` command:
+
+```bash
+zk_inception chain create
+```
+
+This time, use the answers below:
+
+```bash
+$ zk_inception chain create
+
+┌   ZKsync toolbox
+│
+◇  What do you want to name the chain?
+│  custom_zk_chain
+│
+◇  What's the chain id?
+│  272
+│
+◇  Select how do you want to create the wallet
+│  Localhost
+│
+◇  Select the prover mode
+│  NoProofs
+│
+◇  Select the commit data generator mode
+│  Rollup
+│
+◇  Select the base token to use
+│  Custom
+│
+◇  What is the token address?
+│  <0x_YOUR_TOKEN_ADDRESS>
+│
+◇  What is the base token price nominator?
+│  1
+│
+◇  What is the base token price denominator?
+│  1
+│
+◇  Set this chain as default?
+│  Yes
+```
+
+The base token nominator and denominator are used to define the relation of the base token price with ETH.
+For example, if we set the nominator to 20 and the denominator to 1, together the relation is 20/1,
+this would mean that 20 tokens would be given the equivalent value as 1 ETH for gas.
+For testing purposes, we'll just use a 1:1 ratio.
+
+Next, initialize the chain in the ecosystem with:
+
+```bash
+zk_inception chain init
```

Comment:
it won't work our of the box, you have to top up configs/wallets.yaml/governor and chains/<name>/configs/wallets.yaml/governor

Reply:
it seems to work just funding the address in `configs/wallets.yaml/governor`, which I have as a step already

Reply:
Ahh I see that funding the chain governor allows you to deploy the paymaster in the setup. I will update this.